pkg/cmd/infra/router/template.go

```diff
@@ -371,6 +401,42 @@ func (o *TemplateRouterOptions) Run() error {
 		reloadCallbacks = append(reloadCallbacks, collector.CollectNow)
 	}
 
+	kc, err := o.Config.Clients()
+	if err != nil {
+		return err
+	}
+	routeclient, err := routeinternalclientset.NewForConfig(o.Config.OpenShiftConfig())
+	if err != nil {
+		return err
+	}
+	projectclient, err := projectinternalclientset.NewForConfig(o.Config.OpenShiftConfig())
+	if err != nil {
+		return err
+	}
+
+	var cfgManager templateplugin.ConfigManager
+	if o.ConfigManagerName == "haproxy-manager" {
+		blueprintRoutes, err := o.blueprintRoutes(routeclient)
```

Comment:
You're going to have to refresh these, they shouldn't be static or require restart.  Make that dynamic (add a controller plugin wrapper that triggers reload of the blueprints).

Reply:
There's going to be some issues with blueprint refresh that I need to think about - basically vis-a-vis replacement and deletion and how the allocations/config changes, so can we do this in a follow-up PR?